### PR TITLE
adds optional variable for enabling specified autoscaling group metrics

### LINKED ISSUE #98: enable autoscaling group metrics
Metrics appear to be missing from the autoscaling group.

>GroupMinSize, GroupMaxSize, GroupDesiredCapacity, GroupInServiceInstances, GroupPendingInstances, GroupStandbyInstances, GroupTerminatingInstances, GroupTotalInstances

### PR DESCRIPTION
**Pull request by eedwards-sk**
closes #98

### CODE REVIEW

**Comment by eedwards-sk** [comment]
I switched my current project to using this fork/branch to test the change.

I confirmed it's a non-breaking change that supports enabling additional metrics, if specified.

e.g. to my terraform file (that consumes the module) I added:

```hcl
  enabled_metrics = [
    "GroupMinSize",
    "GroupMaxSize",
    "GroupDesiredCapacity",
    "GroupInServiceInstances",
    "GroupPendingInstances",
    "GroupStandbyInstances",
    "GroupTerminatingInstances",
    "GroupTotalInstances"
  ]
```

upon execution against existing cluster, my plan output is:

```
- module.consul_cluster
  Getting source "github.com/eedwards-sk/terraform-aws-consul//modules/consul-cluster?ref=GH-98"

...

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ module.consul_cluster.aws_autoscaling_group.autoscaling_group
      enabled_metrics.#:          "0" => "8"
      enabled_metrics.119681000:  "" => "GroupStandbyInstances"
      enabled_metrics.1940933563: "" => "GroupTotalInstances"
      enabled_metrics.308948767:  "" => "GroupPendingInstances"
      enabled_metrics.3267518000: "" => "GroupTerminatingInstances"
      enabled_metrics.3394537085: "" => "GroupDesiredCapacity"
      enabled_metrics.3551801763: "" => "GroupInServiceInstances"
      enabled_metrics.4118539418: "" => "GroupMinSize"
      enabled_metrics.4136111317: "" => "GroupMaxSize"


Plan: 0 to add, 1 to change, 0 to destroy.
```

and the apply

```
module.consul_cluster.aws_autoscaling_group.autoscaling_group: Modifying... (ID: eedwards-dev-consul-cluster20181112203632309200000008)
  enabled_metrics.#:          "0" => "8"
  enabled_metrics.119681000:  "" => "GroupStandbyInstances"
  enabled_metrics.1940933563: "" => "GroupTotalInstances"
  enabled_metrics.308948767:  "" => "GroupPendingInstances"
  enabled_metrics.3267518000: "" => "GroupTerminatingInstances"
  enabled_metrics.3394537085: "" => "GroupDesiredCapacity"
  enabled_metrics.3551801763: "" => "GroupInServiceInstances"
  enabled_metrics.4118539418: "" => "GroupMinSize"
  enabled_metrics.4136111317: "" => "GroupMaxSize"
module.consul_cluster.aws_autoscaling_group.autoscaling_group: Modifications complete after 2s (ID: eedwards-dev-consul-cluster20181112203632309200000008)

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

File: modules/consul-cluster/main.tf [hunk 1]
```diff
@@ -29,6 +29,8 @@ resource "aws_autoscaling_group" "autoscaling_group" {
   wait_for_capacity_timeout = "${var.wait_for_capacity_timeout}"
   service_linked_role_arn   = "${var.service_linked_role_arn}"
 
+  enabled_metrics = ["${var.enabled_metrics}"]
```
Comment (brikis98) [inline modules/consul-cluster/main.tf]: What is the default value in `aws_autoscaling_group` if you don't specify this variable?

Comment (eedwards-sk) [reply modules/consul-cluster/main.tf]: Great question @brikis98 

The default value is an empty list, which is what I attempted to set the default value for `var.enabled_metrics` to, to match it. Confirmed when I switch back from my module fork (where I specified `enabled_metrics`) to the original unmodified module and run a plan:

```
Initializing modules...
- module.consul_cluster
  Getting source "github.com/hashicorp/terraform-aws-consul//modules/consul-cluster?ref=v0.4.1"
...
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ module.consul_cluster.aws_autoscaling_group.autoscaling_group
      enabled_metrics.#:          "8" => "0"
      enabled_metrics.119681000:  "GroupStandbyInstances" => ""
      enabled_metrics.1940933563: "GroupTotalInstances" => ""
      enabled_metrics.308948767:  "GroupPendingInstances" => ""
      enabled_metrics.3267518000: "GroupTerminatingInstances" => ""
      enabled_metrics.3394537085: "GroupDesiredCapacity" => ""
      enabled_metrics.3551801763: "GroupInServiceInstances" => ""
      enabled_metrics.4118539418: "GroupMinSize" => ""
      enabled_metrics.4136111317: "GroupMaxSize" => ""


Plan: 0 to add, 1 to change, 0 to destroy.
```

Comment (brikis98) [reply modules/consul-cluster/main.tf]: Got it, thanks!

**Comment by brikis98** [comment]
https://github.com/hashicorp/terraform-aws-consul/releases/tag/v0.4.3

**Pull request merged**